### PR TITLE
fix: use current_identity as user's email

### LINKED ISSUE #5802: 'AnonymousUserMixin' object has no attribute 'email' on updating an event
**Describe the bug**

We are facing this Attribute Error the moment we try updating any event related object which corresponds to a celery task

**To Reproduce**
1. Update any existing event
2. See error

**Expected behavior**

It should work normally without any error

**Stacktrace**

```cmd
AttributeError: 'AnonymousUserMixin' object has no attribute 'email'

File "/home/shreyansh/open-event-server/venv/lib/python3.6/site-packages/flask/views.py", line 88, in view
    return self.dispatch_request(*args, **kwargs)
  File "/home/shreyansh/open-event-server/venv/src/flask-rest-jsonapi/flask_rest_jsonapi/resource.py", line 68, in dispatch_request
    response = method(*args, **kwargs)
  File "/home/shreyansh/open-event-server/venv/src/flask-rest-jsonapi/flask_rest_jsonapi/decorators.py", line 56, in wrapper
    return func(*args, **kwargs)
  File "/home/shreyansh/open-event-server/venv/src/flask-rest-jsonapi/flask_rest_jsonapi/resource.py", line 311, in patch
    self._data_layer.update_object(obj, data, kwargs)
  File "/home/shreyansh/open-event-server/venv/src/flask-rest-jsonapi/flask_rest_jsonapi/data_layers/alchemy.py", line 144, in update_object
    self.after_update_object(obj, data, view_kwargs)
  File "/home/shreyansh/open-event-server/app/api/events.py", line 485, in after_update_object
    start_export_tasks(event)
  File "/home/shreyansh/open-event-server/app/api/events.py", line 553, in start_export_tasks
    create_export_job(task_xcal.id, event_id)
```
**Additional details (please complete the following information):**
 - OS: [e.g. MacOS, Ubuntu, CentOS] Ubuntu
 - Python Version [e.g. `3.5`, `3.6`] 3.5

**Additional context**

Working on it

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #5802 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
This makes sure that the user_email being caught is of current_identity instead of current_user

#### Changes proposed in this pull request:

- change current_user with current_identity

### CODE REVIEW

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Please review!

**Comment by mrsaicharan1** [comment]
@uds5501 Fix the failing test case where it checks for `current_user.email` and you'll be good to go!

**Comment by uds5501** [comment]
@mrsaicharan1 Fixed it!

**Comment by uds5501** [comment]
@CosmicCoder96 @iamareebjamal Please review

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 @pradeepgangwar Please review!

File: app/api/helpers/export_helpers.py [hunk 1]
```diff
@@ -241,12 +241,15 @@ def create_export_job(task_id, event_id):
     if export_job:
 
         export_job.task = task_url
-        export_job.user_email = current_user.email
+        if current_identity:
+            export_job.user_email = current_identity.email
+        else:
+            export_job.user_email = current_user.email
```
Comment (iamareebjamal) [inline app/api/helpers/export_helpers.py]: Why this switching?

Comment (uds5501) [reply app/api/helpers/export_helpers.py]: To handle the test case. In the test case, the user is being logged in with the help of flask_login and doesn't use flask_jwt, hence the `current_identity` will be None in that case

Comment (iamareebjamal) [reply app/api/helpers/export_helpers.py]: Create a common method which returns either of them so we can just call `get_current_user().email`

Comment (uds5501) [reply app/api/helpers/export_helpers.py]: @iamareebjamal Sure, on it.

**Pull request merged**